src/material.js

```diff
@@ -184,8 +214,14 @@ export class Material {
      * @param salt {String} Salt for hashing, empty string by default.
      * @param isScaled {Boolean} Whether to scale the lattice parameter 'a' to 1.
      */
-    calculateHash(salt = "", isScaled = false) {
-        const message = `${this.Basis.hashString}#${this.Lattice.getHashString(isScaled)}#${salt}`;
+    calculateHash(salt = '', isScaled = false) {
+        let message = this.Basis.hashString + "#" + this.Lattice.getHashString(isScaled) + "#" + salt;
+        if (this.isNonPeriodic) {
+            const inchiString = this.getDerivedPropertiesForHash();
+            if (inchiString) {
+                message = inchiString;
+            }
+        }
```

Comment:
This logic is masking the problem - if we want to extract the inchi hash, but it's not available, then the hash is still calculated as for the periodic materials. That should not happen, we should throw error.

Reply:
The error check is implemented above https://github.com/Exabyte-io/made.js/pull/47/files#:~:text=Show%20conversation-,if%20(inchi)%20%7B,%7D,-return%20null.

If we try the error here it fails during non-periodic materials creation with the MD because the derived properties have not been created yet.